src/compressor.c

```diff
@@ -146,18 +146,95 @@ size_t compressor_compress(uint32_t c,
 		case COMP_ID_LZ4:
 #ifdef ENABLE_LZ4
 		{
-			int out_bytes;
-
+			size_t out_bytes = 0, max_dest_size = 0;
+			int method = 0;
+#ifdef LZ4HC_CLEVEL_MAX
+			//now we are calling the new lz4 library in Ubuntu 20.04, which is liblz4-1:amd64, 1.9.2-2ubuntu0.20.04.1
+			//calculate the maximal dest size base on the source size
+			max_dest_size = LZ4_compressBound( length );
+			// we allocated using a worst case guess, The actual compressed size might be less than we allocated.
+			// doing so to prevent the API from potential data overflow
+			void *dest_new = calloc(max_dest_size, 1);
 			if (c & COMP_LZ4_HC)
-				out_bytes = LZ4_compressHC_limitedOutput(src, dest, length, out_size);
+			{
+				/*
+LZ4int LZ4_compress_HC
+(
+    const char * src,
+    char * dst,
+    int srcSize,
+    int dstCapacity,
+    int compressionLevel
+)
+: Compress data from src_ into dst, using the powerful but slower "HC" algorithm. 
+`dst must be already allocated. Compression is guaranteed to succeed if dstCapacity >= LZ4_compressBound(srcSize)_ (see "lz4.h") 
+Max supported srcSize value is LZ4_MAX_INPUT_SIZE (see "lz4.h") 
+`compressionLevel : any value between 1 and LZ4HC_CLEVEL_MAX will work. Values > LZ4HC_CLEVEL_MAX behave the same as LZ4HC_CLEVEL_MAX.
+Returns: the number of bytes written into 'dst' or 0 if compression fails.
+				*/
+				out_bytes = LZ4_compress_HC(src, dest_new, length, max_dest_size, LZ4HC_CLEVEL_MAX);
+				method = 1;
+				fprintf(stderr, "20.04 LZ4_compress_HC: source size: %lu, max_dest_size: %lu, out_bytes:%lu, expected size: %lu\n", length, max_dest_size, out_bytes, out_size);
```

Comment:
I can't say the original file is 100% consistent but keep below 80 cols. Also you use spaces inconsistently. Also `20.04` is entirely meaningless to the end user, and this whole thing looks like debug, not error output.